### PR TITLE
Non desired window opening alongside ignition GUI

### PR DESCRIPTION
**Pull request by ahcorde**
# 🦟 Bug fix

## Summary

Fix https://github.com/ignitionrobotics/ign-gui/issues/291

Requires a PR in ign-gui6 https://github.com/ignitionrobotics/ign-gui/pull/292

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

File: include/ignition/rendering/RenderEngine.hh [hunk 1]
```diff
@@ -172,6 +172,15 @@ namespace ignition
       /// \return True if headless mode is enable, false otherwise.
       public: virtual bool Headless() const = 0;
 
+      /// \brief Set winID from Qt
+      /// Required in OGRE 2.2
+      /// \param[in] _winID Qt MainWindow ID
```
Comment (chapulina) [inline include/ignition/rendering/RenderEngine.hh:177]: Can we make the documentation less Ogre / Qt specific? Could this function be useful for other engines and / or GUI frameworks?

**Comment by ahcorde** [comment]
> I think this new `SetWinID` API is no longer needed because winId is now passed as one of the engine params in [ignitionrobotics/ign-gui#292](https://github.com/ignitionrobotics/ign-gui/pull/292)?

These changes are required because the params passed by `ign-gui` are parsed inside `ign-rendering` to offer the API to the different Render engines

**Comment by chapulina** [comment]
> These changes are required because the params passed by ign-gui are parsed inside ign-rendering to offer the API to the different Render engines

I'm not sure I understand. Can you point out where `SetWinId` is used?

**Comment by matosinho** [comment]
Issue fixed here for me with this branch.

**Comment by matosinho** [comment]
Issue fixed here for me with this branch.

**Comment by ahcorde** [comment]
> I'm not sure I understand. Can you point out where `SetWinId` is used?

The `params` variable is the same type in `ign-gui` and `ign-rendering` but both libraries need to keep it own copy.

**Review by chapulina — APPROVED** [review]
Works for me :+1:

**Pull request merged**